File: app/src/main/java/mariavv/fitnesspal/domain/Food.java
package mariavv.fitnesspal.domain;

import android.support.annotation.NonNull;

public class Food {

    private static final int PER_WEIGHT = 100;
    private static final int PROTEIN_ENERGY = 4;
    private static final int FAT_ENERGY = 9;
    private static final int CARB_ENERGY = 4;

    @NonNull
    private String name;
    private int protein;
    private int fat;
    private int carb;

    public Food(@NonNull String name, int protein, int fat, int carb) {
        this.name = name;
        this.protein = protein;
        this.fat = fat;
        this.carb = carb;
    }

    public int getEnergy(int weight) {
        return (protein * PROTEIN_ENERGY
                + fat * FAT_ENERGY
                + carb * CARB_ENERGY)
                * weight / PER_WEIGHT;
    }

    @NonNull
    public String getName() {
        return name;
    }

    public int getProtein() {
        return protein;
    }

    public int getFat() {
        return fat;
    }

    public int getCarb() {
        return carb;
    }

    public int getCount(int macronutrient, int weight) {
        return macronutrient * weight / PER_WEIGHT;
    }
}

File: app/src/main/java/mariavv/fitnesspal/domain/Food.kt
package mariavv.fitnesspal.domain

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey
import mariavv.fitnesspal.data.db.ColumnName
import mariavv.fitnesspal.data.db.TableName

@Entity(tableName = TableName.FOODS)
class Food(
        @PrimaryKey(autoGenerate = true)
        @ColumnInfo(name = ColumnName.ID)
        var id: Int = 0,
        @ColumnInfo(name = ColumnName.NAME)
        var name: String,
        @ColumnInfo(name = ColumnName.PROTEIN)
        var protein: Int,
        @ColumnInfo(name = ColumnName.FAT)
        var fat: Int,
        @ColumnInfo(name = ColumnName.CARB)
        var carb: Int,
        @ColumnInfo(name = ColumnName.SORTABLE_NAME)
        var sortableName: String = "") {

    fun getEnergy(weight: Int): Int {
        return (protein * PROTEIN_ENERGY
                + fat * FAT_ENERGY
                + carb * CARB_ENERGY) * weight / PER_WEIGHT
    }

    companion object {
        private val PER_WEIGHT = 100
        private val PROTEIN_ENERGY = 4
        private val FAT_ENERGY = 9
        private val CARB_ENERGY = 4

        fun getMacronutrientInWeight(macronutrient: Int, weight: Int): Int {
            return macronutrient * weight / PER_WEIGHT
        }
    }

    constructor() : this(name = "", protein = 0, fat = 0, carb = 0)
}
